spec.bs

```diff
@@ -234,6 +234,34 @@ Browsing contexts -- that are in the background and have their documents in <a h
 When a [=top-level browsing context=] (tab in the browser) is [=/discarded=] due to resource pressure (or unexpected events eg. process crash), and later the user revisits the tab in the browser, then the {{Document}}'s [=Document/discarded=] boolean will be true due to [[#html-initialize-doc]].
 
 
+Feature Policies {#feature-policies}
+=====================
+To improve CPU utilization of sub-frames that are not critical to a page's usability two feature policies are defined:
+ * execution-while-not-rendered, which has a <a href="https://wicg.github.io/feature-policy/#default-allowlist">default allowlist</a> of '*'
+ * execution-while-out-of-viewport, which has a <a href="https://wicg.github.io/feature-policy/#default-allowlist">default allowlist</a> of '*'
+
+<dfn export lt="execution-while-not-rendered">execution-while-not-rendered</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for frames that are not part of rendering. eg. `display: none`.
+
+<dfn export lt="execution-while-out-of-viewport">execution-while-out-of-viewport</dfn> policy controls whether [=task source/freezable=]
+tasks should execute for frames that are scrolled out of the current visible viewport.
+
+Application of these policies is accomplished by placing the frame (and its decendants) in a [=/frozen=] state when the following
+conditions have been met:
+ * [=iframe-load-event-steps] have been run
+ * the policy is disabled for the frame
+ * relevant policy condition applies (not rendered or scrolled out of view)
+
+If these conditions aren't met the frame should be in the [=/unfrozen=] state.
```

Comment:
So concretely it sounds like this would be a modification to the iframe load event steps, which consults the policy and then sets the frame's document to the frozen state. Could you model this as a patch in that way, e.g. insert the following step between steps Y and step Z? That would be clearer than this action-at-a-distance phrasing.

Reply:
Done